### PR TITLE
Remove Greenplum-only logic transformGroupedWindows.

### PR DESCRIPTION
**Pull request by kainwen**
The function `transformGroupedWindows ` was added about 10 years
ago to fix some jira. Postgres never has such logic. This commit removes
the whole logic and gets positive results:
  * Some cases before this patch do not give correct results
    and now they run correctly (e.g. case `regress/olap_window_seq`)
  * The Github Issue https://github.com/greenplum-db/gpdb/issues/10143
    is fixed by this commit.

----------

**I have verified every test cases change by comparing postgres's result and gpdb(with this patch).  GPDB and Postgres give the same results.**

For the cause of the github issue, please refer to the details at Github Issue https://github.com/greenplum-db/gpdb/issues/10143.

To be honest, I do not quite understand the logic of `transformGroupedWindows`. I run the full pipeline, it turns green.

--------------

I also cherry-pick this to 6X to see if it is OK. On 6X, it does not assert fail but `ERROR:  variable not found in subplan target list (setrefs.c:2489)`.

### CODE REVIEW

File: src/test/regress/expected/window_optimizer.out [hunk 1]
```diff
@@ -631,13 +631,13 @@ select first_value(max(x)) over (), y
...
                ->  Redistribute Motion 3:3  (slice2; segments: 3)
-                     Hash Key: ((ten + four))
-                     ->  Streaming Partial HashAggregate
-                           Group Key: (ten + four)
+                     Hash Key: ((tenk1.ten + tenk1.four))
+                     ->  Partial HashAggregate
+                           Group Key: (tenk1.ten + tenk1.four)
                            ->  Seq Scan on tenk1
- Optimizer: Pivotal Optimizer (GPORCA) version 3.83.0
+ Optimizer: Postgres query optimizer
```
Comment (hlinnaka) [inline src/test/regress/expected/window_optimizer.out:640]: Looks like ORCA can't create a plan for this anymore, and falls back. Is that acceptable? Or can we fix ORCA to deal with that easily?

Comment (kainwen) [reply src/test/regress/expected/window_optimizer.out:640]: > Looks like ORCA can't create a plan for this anymore, and falls back. Is that acceptable? Or can we fix ORCA to deal with that easily?

The icw planner is green however icw orca fails for just the same reason as this:

```
--- /tmp/build/e18b2f02/gpdb_src/src/test/regress/expected/gporca_optimizer.out	2020-05-19 13:02:36.094534377 +0000
+++ /tmp/build/e18b2f02/gpdb_src/src/test/regress/results/gporca.out	2020-05-19 13:02:36.598569248 +0000
@@ -5168,6 +5172,8 @@
 (40 rows)
 
 select rank() over(partition by b order by count(*)/sum(a)) from orca.foo group by a, b order by 1;
+INFO:  GPORCA failed to produce a plan, falling back to planner
+DETAIL:  Query-to-DXL Translation: No attribute entry found due to incorrect normalization of query
  rank 
 ------
     1
@@ -5213,6 +5219,8 @@
 (40 rows)
 
 select row_number() over(order by foo.a) from orca.foo inner join orca.bar using(b) group by foo.a, bar.b, bar.a;
+INFO:  GPORCA failed to produce a plan, falling back to planner
+DETAIL:  CTranslatorQueryToDXL.cpp:597: Failed assertion: __null == m_query->groupClause
  row_number 
 ------------
           1
@@ -6023,6 +6031,8 @@
 (400 rows)
 
 select row_number() over(order by foo.a+ bar.a)/count(*) from orca.foo inner join orca.bar using(b) group by foo.a, bar.a, bar.b;
+INFO:  GPORCA failed to produce a plan, falling back to planner
+DETAIL:  Query-to-DXL Translation: No attribute entry found due to incorrect normalization of query
  ?column? 
 ----------
         1
@@ -6689,6 +6699,8 @@
 
 -- aggref in subquery with window func in target list
 select orca_w1.a, (select sum(orca_w2.a) from orca_w2 where orca_w1.b = orca_w2.b), count(*), rank() over (order by orca_w1.b) from orca_w1 group by orca_w1.a, orca_w1.b order by orca_w1.a;
+INFO:  GPORCA failed to produce a plan, falling back to planner
+DETAIL:  Query-to-DXL Translation: No attribute entry found due to incorrect normalization of query
  a | sum | count | rank 
 ---+-----+-------+------
  1 |     |     1 |    1

======================================================================
```

I think we'd better look into this to find out why orca falls back.

**Comment by vraghavan78** [comment]
@chrishajas @asubramanya @zellerh @ashuka24 @hardikar can you please take a look at this early next week?

**Comment by zellerh** [comment]
I'm taking a look at the fallback.

**Comment by vraghavan78** [comment]
@kainwen it seems quite a few TPC-DS queries are now falling back. Please do not merge this till we have answers / solutions from the ORCA team. Thanks!

V

**Comment by hlinnaka** [comment]
Could we move the call so that it's only done before sending the query to ORCA?

**Comment by kainwen** [comment]
Since @hlinnaka opens a pr https://github.com/greenplum-db/gpdb/pull/10306
can both:
  * fix the issue
  * prevent orca fallback

So close this one.

**Comment by hlinnaka** [comment]
> Could we move the call so that it's only done before sending the query to ORCA?

I opened https://github.com/greenplum-db/gpdb/pull/10306 to do that. Please take a look.

I really wish this was fixed on ORCA's side, but this at least lets us move forward with the Postgres planner.

**Pull request closed without merging**